shared/js/ui/models/grade-scorecard.es6.js

```diff
@@ -0,0 +1,75 @@
+const Parent = window.DDG.base.Model
+
+function GradeScorecard (attrs) {
+  Parent.call(this, attrs)
+
+  this.reasons = this._getReasons()
+}
+
+GradeScorecard.prototype = window.$.extend({},
+  Parent.prototype,
+  {
+    modelName: 'gradeScorecard',
+
+    _getReasons: function () {
+      const site = this.site
```

Comment:
I don't get why you would rewire `site` up as a `const` here instead of just checking `this.site` throughout the function. If we bulk up the models with cruft like this they will become difficult to refactor later on.

Reply:
I've generally done this to reduce the number of times I have to type `this.`, since it's used multiple times in the function, plus it allows us to shave a few bytes off in minifiable code. Whether it makes refactoring more difficult or not largely depends on what kind of refactoring you need to do, e.g. if you change how `site` is accessed you may only need to edit one line.

I'm not strongly attached to the way I did it. I'll update the code to use `this.site`.